Add auth unique identifier support

thegcat opened the pull request:
Fixes #8

thegcat committed:
Add auth unique identifier support #8

Sohalt commented on src/pretix_ldap/ldap_connector.py:
```diff
@@ -75,15 +78,15 @@ def form_authenticate(self, request, form_data):
+            )
+        except EmailAddressTakenError:
+            messages.error(
+                request, _('We cannot create your user account as a user account in this system '
```
I don't know how I feel about this error message, since it leaks information about the registered users. Maybe just log the error and show a more generic message?

raphaelm replied:
It most likely only leaks information to someone who controls exactly that email address, unless the LDAP contains unverified email addresses

Sohalt replied:
It can be used find out if a user is registered at a specific pretix instance, can't it? I.e. I can start to enumerate email addresses and find out for which of them an account exists.

raphaelm replied:
Yeah but for every enumeration step you need to do a valid ldap login for the address you are trying to enumerate if I'm not mistaken

thegcat replied:
Yes, the error message would only appear if you have a valid login for the LDAP, i.e. an enumeration of already "used" email addresses would be possible but only if you already have working LDAP credentials.

thegcat replied:
I have removed for now though, so feel free to handle this as you think is best.

Sohalt replied:
Sorry, I didn't thoroughly look at this before writing that comment. Given the constraint, that you already have to have a valid login, I think the more explicit error message is fine.

thegcat committed:
Support and test with dn as the unique attribute

thegcat committed:
Remove double slashes in test URLs

thegcat commented:
The last commit [c8dc11e](https://github.com/Sohalt/pretix-ldap/pull/9/commits/c8dc11ecb223b028da37c6980ebf7eb84291e5cd) was required to get the tests/docker to work on my machine, I don't know if this would be required elsewhere.

If you are happy with the PR I would squash all commits except the last and adapt the Pretix requirement accordingly in the PretixPluginMeta and mark the PR as ready.

Sohalt commented:
Sorry for taking so long to respond. Thank you a lot for your contribution. I'm fine with merging this after you squash. Also feel free to add back the more informative error message.

thegcat committed:
Version 0.1.0

thegcat commented:
I have rebased and squashed a few commits, this should be mergeable as is. Feel free to poke in the commits if you’d like to have them arranged another way.
